Stopped working (compilation failure) after Linux 5.18.0-arch1-1 update

I installed the latest Arch Linux kernel a few days ago and noticed the modem wasn't starting up anymore.
After some troubleshooting, I figured out compilation now seems to fail.
I can't assure it is linked to the kernel update, but it happened right after it so I guess it's probable.
```bash
# sudo systemctl status xmm7360.service
× xmm7360.service - XMM7360 Modem Init
     Loaded: loaded (/usr/lib/systemd/system/xmm7360.service; enabled; vendor preset: disabled)
     Active: failed (Result: exit-code) since Sat 2022-05-28 10:01:02 CEST; 10min ago
    Process: 41718 ExecStart=/usr/lib/xmm7360-pci-git/rpc/open_xdatachannel.py -c /etc/xmm7360 (code=exited, status=1/FAILURE)
   Main PID: 41718 (code=exited, status=1/FAILURE)
        CPU: 199ms

May 28 10:01:01 carbon systemd[1]: Starting XMM7360 Modem Init...
May 28 10:01:02 carbon open_xdatachannel.py[41718]: Traceback (most recent call last):
May 28 10:01:02 carbon open_xdatachannel.py[41718]:   File "/usr/lib/xmm7360-pci-git/rpc/open_xdatachannel.py", line 42, in <module>
May 28 10:01:02 carbon open_xdatachannel.py[41718]:     r = rpc.XMMRPC()
May 28 10:01:02 carbon open_xdatachannel.py[41718]:   File "/usr/lib/xmm7360-pci-git/rpc/rpc.py", line 19, in __init__
May 28 10:01:02 carbon open_xdatachannel.py[41718]:     self.fp = os.open(path, os.O_RDWR | os.O_SYNC)
May 28 10:01:02 carbon open_xdatachannel.py[41718]: FileNotFoundError: [Errno 2] No such file or directory: '/dev/xmm0/rpc'
May 28 10:01:02 carbon systemd[1]: xmm7360.service: Main process exited, code=exited, status=1/FAILURE
May 28 10:01:02 carbon systemd[1]: xmm7360.service: Failed with result 'exit-code'.
May 28 10:01:02 carbon systemd[1]: Failed to start XMM7360 Modem Init.
```
```bash
# yay -S xmm7360-pci-dkms-git
:: Checking for conflicts...
:: Checking for inner conflicts...
[Aur:1]  xmm7360-pci-git-r229.cf6625a-1 (xmm7360-pci-dkms-git)

  1 xmm7360-pci-git (xmm7360-pci-dkms-git)   (Installed) (Build Files Exist)
==> Packages to cleanBuild?
==> [N]one [A]ll [Ab]ort [I]nstalled [No]tInstalled or (1 2 3, 1-3, ^4)
==> 
:: PKGBUILD up to date, Skipping (1/0): xmm7360-pci-git
  1 xmm7360-pci-git (xmm7360-pci-dkms-git)   (Installed) (Build Files Exist)
==> Diffs to show?
==> [N]one [A]ll [Ab]ort [I]nstalled [No]tInstalled or (1 2 3, 1-3, ^4)
==> 
:: (1/1) Parsing SRCINFO: xmm7360-pci-git (xmm7360-pci-dkms-git)
==> Making package: xmm7360-pci-git r229.cf6625a-1 (Sat 28 May 2022 10:04:05 CEST)
==> Retrieving sources...
  -> Updating xmm7360-pci git repo...
Fetching origin
  -> Found dkms.conf
  -> Found xmm7360.service
  -> Found nodbus-exit-code.patch
  -> Found dns-priority.patch
==> Validating source files with sha256sums...
    xmm7360-pci ... Skipped
    dkms.conf ... Passed
    xmm7360.service ... Passed
    nodbus-exit-code.patch ... Passed
    dns-priority.patch ... Passed
==> Making package: xmm7360-pci-git r229.cf6625a-1 (Sat 28 May 2022 10:04:07 CEST)
==> Checking runtime dependencies...
==> Checking buildtime dependencies...
==> Retrieving sources...
  -> Updating xmm7360-pci git repo...
Fetching origin
  -> Found dkms.conf
  -> Found xmm7360.service
  -> Found nodbus-exit-code.patch
  -> Found dns-priority.patch
==> Validating source files with sha256sums...
    xmm7360-pci ... Skipped
    dkms.conf ... Passed
    xmm7360.service ... Passed
    nodbus-exit-code.patch ... Passed
    dns-priority.patch ... Passed
==> Removing existing $srcdir/ directory...
==> Extracting sources...
  -> Creating working copy of xmm7360-pci git repo...
Cloning into 'xmm7360-pci'...
done.
==> Starting prepare()...
patching file rpc/open_xdatachannel.py
patching file rpc/open_xdatachannel.py
==> Starting pkgver()...
==> Sources are ready.
==> Making package: xmm7360-pci-git r229.cf6625a-1 (Sat 28 May 2022 10:04:10 CEST)
==> Checking runtime dependencies...
==> Checking buildtime dependencies...
==> WARNING: Using existing $srcdir/ tree
==> Starting pkgver()...
==> Sources are ready.
 -> xmm7360-pci-git-r229.cf6625a-1 already made -- skipping build
 -> Found git repo: github.com/xmm7360/xmm7360-pci
loading packages...
warning: xmm7360-pci-dkms-git-r229.cf6625a-1 is up to date -- reinstalling
resolving dependencies...
looking for conflicting packages...

Packages (1) xmm7360-pci-dkms-git-r229.cf6625a-1

Total Installed Size:  0.04 MiB
Net Upgrade Size:      0.00 MiB

:: Proceed with installation? [Y/n] 
(1/1) checking keys in keyring                                                                                  [###################################################################] 100%
(1/1) checking package integrity                                                                                [###################################################################] 100%
(1/1) loading package files                                                                                     [###################################################################] 100%
(1/1) checking for file conflicts                                                                               [###################################################################] 100%
(1/1) checking available disk space                                                                             [###################################################################] 100%
:: Running pre-transaction hooks...
(1/1) Remove upgraded DKMS modules
==> dkms remove xmm7360-pci-git/r229.cf6625a
:: Processing package changes...
(1/1) reinstalling xmm7360-pci-dkms-git                                                                         [###################################################################] 100%
:: Running post-transaction hooks...
(1/2) Arming ConditionNeedsUpdate...
(2/2) Install DKMS modules
==> dkms install --no-depmod xmm7360-pci-git/r229.cf6625a -k 5.18.0-arch1-1
Error! Bad return status for module build on kernel: 5.18.0-arch1-1 (x86_64)
Consult /var/lib/dkms/xmm7360-pci-git/r229.cf6625a/build/make.log for more information.
==> WARNING: `dkms install --no-depmod xmm7360-pci-git/r229.cf6625a -k 5.18.0-arch1-1' exited 10
```
```bash
# cat /var/lib/dkms/xmm7360-pci-git/r229.cf6625a/build/make.log
DKMS make.log for xmm7360-pci-git-r229.cf6625a for kernel 5.18.0-arch1-1 (x86_64)
Sat 28 May 10:04:12 CEST 2022
make -C /usr/lib/modules/5.18.0-arch1-1/build M=/var/lib/dkms/xmm7360-pci-git/r229.cf6625a/build modules
make[1]: Entering directory '/usr/lib/modules/5.18.0-arch1-1/build'
  CC [M]  /var/lib/dkms/xmm7360-pci-git/r229.cf6625a/build/xmm7360.o
/var/lib/dkms/xmm7360-pci-git/r229.cf6625a/build/xmm7360.c: In function ‘xmm7360_irq0’:
/var/lib/dkms/xmm7360-pci-git/r229.cf6625a/build/xmm7360.c:1175:13: warning: the comparison will always evaluate as ‘true’ for the address of ‘td_ring’ will never be NULL [-Waddress]
 1175 |         if (xmm->td_ring) {
      |             ^~~
/var/lib/dkms/xmm7360-pci-git/r229.cf6625a/build/xmm7360.c:209:24: note: ‘td_ring’ declared here
  209 |         struct td_ring td_ring[16];
      |                        ^~~~~~~
/var/lib/dkms/xmm7360-pci-git/r229.cf6625a/build/xmm7360.c: In function ‘xmm7360_probe’:
/var/lib/dkms/xmm7360-pci-git/r229.cf6625a/build/xmm7360.c:1455:15: error: implicit declaration of function ‘pci_set_dma_mask’ [-Werror=implicit-function-declaration]
 1455 |         ret = pci_set_dma_mask(dev, 0xffffffffffffffff);
      |               ^~~~~~~~~~~~~~~~
cc1: some warnings being treated as errors
make[2]: *** [scripts/Makefile.build:288: /var/lib/dkms/xmm7360-pci-git/r229.cf6625a/build/xmm7360.o] Error 1
make[1]: *** [Makefile:1834: /var/lib/dkms/xmm7360-pci-git/r229.cf6625a/build] Error 2
make[1]: Leaving directory '/usr/lib/modules/5.18.0-arch1-1/build'
make: *** [Makefile:9: default] Error 2
```
```bash
# uname -rs
Linux 5.18.0-arch1-1
```
Note: I'm on a ThinkPad X1 Carbon 7th gen.

--- a/xmm7360.c
+++ b/xmm7360.c
@@ -1452,12 +1452,11 @@ static int xmm7360_probe(struct pci_dev *dev, const struct pci_device_id *id)
 	}
 	pci_set_master(dev);
 
-	ret = pci_set_dma_mask(dev, 0xffffffffffffffff);
+	ret = dma_set_mask_and_coherent(xmm->dev, DMA_BIT_MASK(64));
 	if (ret) {
 		dev_err(xmm->dev, "Cannot set DMA mask\n");
 		goto fail;
 	}
-	dma_set_coherent_mask(xmm->dev, 0xffffffffffffffff);
 
 	ret = pci_request_region(dev, 0, "xmm0");
 	if (ret) {

